### PR TITLE
frontend test( utils+ signup request related test+ post related post)

### LINKED ISSUE #158: Frontend Unit Tests

### PR DESCRIPTION
**Pull request by JialinX**
More tests to be added

### CODE REVIEW

**Review by PENGYUXIONG — COMMENTED** [review]
Do we need `chromedriver` and `chromedriver.exe` to run tests?

**Comment by JialinX** [comment]
> Do we need `chromedriver` and `chromedriver.exe` to run tests?

Yes, chromedriver is required to open and control chrome to run our frontend tests

File: frontend/test/readme.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+installation
+===================================
+```
+pip install selenium
```
Comment (PENGYUXIONG) [inline frontend/test/readme.md:4]: put this in the requirement.txt, so that it will get installed every time by default

**Review by PENGYUXIONG — COMMENTED** [review]
as chromedriver.exe is most likely for windows, if it is generate from selenium we dont need to have it in code base.

**Comment by JialinX** [comment]
> as chromedriver.exe is most likely for windows, if it is generate from selenium we dont need to have it in code base.

Both chromedriver and chromedriver.exe are not generated, I downloaded them from https://sites.google.com/a/chromium.org/chromedriver/home

**Review by PENGYUXIONG — APPROVED** [review]
If all tests are processed correctly and passing then feel free to merge

**Pull request merged**